bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/TriggerLocationCompactService.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.bookkeeper.server.http.service;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.bookkeeper.common.util.JsonUtil;
+import org.apache.bookkeeper.http.HttpServer;
+import org.apache.bookkeeper.http.service.HttpEndpointService;
+import org.apache.bookkeeper.http.service.HttpServiceRequest;
+import org.apache.bookkeeper.http.service.HttpServiceResponse;
+import org.apache.bookkeeper.proto.BookieServer;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * HttpEndpointService that handle force trigger entry location compact requests.
+ *
+ * <p>The PUT method will trigger entry location compact on current bookie.
+ *
+ * <p>The GET method will get the entry location compact running or not.
+ * Output would be like:
+ *        {
+ *           "is_entry_location_compact" : "false"
+ *        }
+ */
+
+public class TriggerLocationCompactService implements HttpEndpointService {
+
+    static final Logger LOG = LoggerFactory.getLogger(TriggerLocationCompactService.class);
+
+    private final BookieServer bookieServer;
+
+    public TriggerLocationCompactService(BookieServer bookieServer) {
+        this.bookieServer = checkNotNull(bookieServer);
+    }
+
+    @Override
+    public HttpServiceResponse handle(HttpServiceRequest request) throws Exception {
+        HttpServiceResponse response = new HttpServiceResponse();
+
+        if (HttpServer.Method.PUT.equals(request.getMethod())) {
+            String requestBody = request.getBody();
+            String output = "Not trigger Entry Location RocksDB compact.";
+            if (StringUtils.isNotBlank(requestBody)) {
+                @SuppressWarnings("unchecked")
+                Map<String, Object> configMap = JsonUtil.fromJson(requestBody, HashMap.class);
```

Comment:
If the JsonUtil.fromJson throws an exception, do we need to catch this exception and return an error response to the client?

Reply:
addressed

Reply:
We would better catch Exception.